### PR TITLE
Dynamic fees

### PR DESCRIPTION
**Pull request by EvgenKor**
#### The main changes:
- adapted to the dynamic fee library version (https://github.com/zkBob/zkbob-client-js/pull/130)
- `tx-amounts` command now gets just amounts list (fee argument was removed)
- changed output for `fee-estimate-....` and `max-transfer` commands to support dynamic fee components
- the default fee rounding config (number of digits after the decimal point in fee estimation) made as following: `BOB-sepolia`: 2, `BOB-goerli`: 1, `BOB-op-goerli`: no rounding (made just for testing and demonstration)

### CODE REVIEW

File: src/account.ts [hunk 1]
```diff
@@ -693,10 +703,11 @@ export class Account {
         console.log('Waiting while state become ready...');
         const ready = await this.getZpClient().waitReadyToTransact();
         if (ready) {
-            const txFee = (await this.getZpClient().feeEstimate([amount], TxType.Transfer, false));
+            const relayerFee = await this.getZpClient().getRelayerFee();
+            console.info(`Using relayer fee: base = ${relayerFee.fee}, perByte = ${relayerFee.oneByteFee}`);
 
             console.log('Making withdraw...');
-            const jobIds: string[] = await this.getZpClient().withdrawMulti(address, amount, txFee.totalPerTx);
+            const jobIds: string[] = await this.getZpClient().withdrawMulti(address, amount, relayerFee);
```
Comment (AllFi) [inline src/account.ts]: It appears that the branch in `zkbob-client-js` includes support for swapping to native tokens, while this branch does not. The third argument in `withdrawMulti` should be `swapAmount` to work properly.

Comment (EvgenKor) [reply src/account.ts]: Yes, I forgot to merge native amount PR here. Now it should work

File: README.md [hunk 2]
```diff
@@ -91,11 +91,11 @@ Suppose you already done local running and set appropriated parameters and setti
 
 `max-transfer` calculate maximum available token amount for outcoming transaction (transfer or withdrawal)
 
-`tx-amounts <amount> [fee] [+]`, get transfer\withdrawal configuration for specified amount ant fee per transaction. Type `+` sign at the end of the command to enter the multitransfer mode
+`tx-amounts <amount> [amount2 amount3 ...]`, get transfer or withdrawal (will requested additionally) configuration for specified amount ant fee per transaction. Type `+` sign at the end of the command to enter the multitransfer mode
```
Comment (AllFi) [inline README.md]:
```suggestion
`tx-amounts <amount> [amount2 amount3 ...]`, get transfer or withdrawal (will requested additionally) configuration for specified amount and fee per transaction. Type `+` sign at the end of the command to enter the multitransfer mode
```

File: src/commands.ts [hunk 3]
```diff
@@ -191,42 +191,45 @@ export async function approveToken(spender: string, amount: string) {
...
+    const txName = txType == TxType.Transfer ? 'transfer' : 'withdraw';
+    
     this.pause();
-    const result: TransferConfig[] = await this.account.getTxParts(amounts, actualFee);
+    const result: TransferConfig[] = await this.account.getTxParts(txType, amountsBN);
     this.resume();
 
     if (amounts.length > 1) {
-        this.echo(`Multi-destination request: ${ amounts.map(async a => `^${await this.account.shieldedToHuman(a)}`).join(', ') }`);
+        this.echo(`Multi-destination request: ${ amountsBN.map(async a => `^${await this.account.shieldedToHuman(a)}`).join(', ') }`);
```
Comment (AllFi) [inline src/commands.ts]: It seems that we don't await promises here:
image

Comment (EvgenKor) [reply src/commands.ts]: Fixed in https://github.com/zkBob/zkbob-console/pull/82/commits/eabc4ea94ca520fe321b4e25e7d15d8553a79162

**Review by AllFi — APPROVED** [review]
It works properly and looks good to me.

**Review by k1rill-fedoseev — APPROVED** [review]

**Pull request merged**